src/core/probe/kernel/bpf/include/common.h

```diff
@@ -210,10 +210,20 @@ static __always_inline int chain(struct retis_context *ctx)
 	 * event usage length read before and after the hook chain.
 	 */
 	struct common_task_event *ti;
+	static bool enabled = false;
 	volatile u16 pass_threshold;
 	struct common_event *e;
 	struct kernel_event *k;
 
+	/* Check if the collection is enabled, otherwise bail out. Once we have
+	 * a positive result, cache it.
+	 */
+	if (!enabled) {
```

Comment:
what about adding this in a common header (common_defs.h)?:

```c
#ifndef likely
#define likely(x)	__builtin_expect(!!(x), 1)
#endif

#ifndef unlikely
#define unlikely(x)	__builtin_expect(!!(x), 0)
#endif
```

and use it here?

Reply:
I think that makes sense; if that indeed makes a difference when building BPF programs (it seems some are using that so I guess so).

Reply:
it makes a difference in the build